internal/cli/service.go

```diff
@@ -663,6 +670,358 @@ func (s Service) findLeafReferences(files []string) (map[string]map[string][]str
 	return matches, nil
 }
 
+type baseLayerSpec struct {
+	Os   string `yaml:"os"`
+	Tag  string `yaml:"tag"`
+	Arch string `yaml:"arch"`
+}
+
+func (b baseLayerSpec) Merge(other baseLayerSpec) baseLayerSpec {
+	os := b.Os
+	if other.Os != "" {
+		os = other.Os
+	}
+
+	tag := b.Tag
+	if other.Tag != "" {
+		tag = other.Tag
+	}
+
+	arch := b.Arch
+	if other.Arch != "" {
+		arch = other.Arch
+	}
+
+	return baseLayerSpec{
+		Os:   os,
+		Tag:  tag,
+		Arch: arch,
+	}
+}
+
+type baseLayerRunFile struct {
+	Spec     baseLayerSpec
+	Filepath string
+	Error    error
+	Updated  bool
+}
+
+var reTasks = regexp.MustCompile(`(?m)^tasks:`)
```

Comment:
I should've caught this on the first pass but technically JSON is valid YAML so "tasks": would be valid too. Not sure if we should support that or just make sure it gives a very explicit error

Reply:
Hmm, I think we may just want to not support that for now? Just quotes is fine, but presumably you could have a yaml file that uses json to cram everything onto a single line, and that seems a bridge too far for this.

Reply:
It won't be an error, because we won't know that it's a run definition - it'll miss the regexp and either: (1) print that no definitions were found or (2) print that nothing needed to be updated. We just won't ever try to update it.